app/main.py

```diff
@@ -1,34 +1,98 @@
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(self, row: int, column: int, is_alive: bool = False) -> None:
+        self.row = row
+        self.column = column
+
+    def __eq__(self, other: tuple) -> bool:
+        return self.row == other[0] and self.column == other[1]
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
+    def __init__(
+            self,
+            start: tuple,
+            end: tuple,
+            is_drowned: bool = False
+    ) -> None:
         # Create decks and save them to a list `self.decks`
-        pass
+        self.decks = []
+        self.is_drowned = is_drowned
+        self.alive_decks = 0
+
+        for row in range(start[0], end[0] + 1):
+            for column in range(start[1], end[1] + 1):
+                self.decks.append(Deck(row, column))
+                self.alive_decks += 1
 
-    def get_deck(self, row, column):
+    def get_deck(self, row: int, column: int) -> Deck:
         # Find the corresponding deck in the list
-        pass
+        deck_index = self.decks.index((row, column))
+        return self.decks[deck_index]
 
-    def fire(self, row, column):
+    def fire(self, row: int, column: int) -> None:
         # Change the `is_alive` status of the deck
         # And update the `is_drowned` value if it's needed
-        pass
+        self.get_deck(row, column).is_alive = False
+        self.alive_decks -= 1
+        if self.alive_decks == 0:
+            self.is_drowned = True
 
 
 class Battleship:
-    def __init__(self, ships):
+    def __init__(self, ships: list[tuple]) -> None:
         # Create a dict `self.field`.
         # Its keys are tuples - the coordinates of the non-empty cells,
         # A value for each cell is a reference to the ship
         # which is located in it
-        pass
+        self.field = {}
+        for coordinates in ships:
+            self.field[coordinates] = Ship(coordinates[0], coordinates[1])
```

Comment:
# Its keys are tuples - the coordinates of the non-empty cells,
But what about decks?